### PR TITLE
Feature/add aws providerconfig

### PR DESCRIPTION
**Pull request by rifisdfds**
- Adds the creation of a ProviderConfig if annotations are set
- Allows choice of annotation name to enable dfds-crossplane and retrieve aws account number

### CODE REVIEW

File: controllers/namespace_controller.go [hunk 1]
```diff
@@ -75,14 +75,39 @@ func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+		log.Log.Info("AWS Account ID value found: " + namespace.Annotations[getAWSAccountIDAnnotationName])
+		awsAccountId = namespace.Annotations[getAWSAccountIDAnnotationName]
+	}
+
+	//configMap := getConfigMap("DFDS_CROSSPLANE_CONFIGMAP_NAME", "DFDS_CROSSPLANE_CONFIGMAP_NAMESPACE")
+
+	// TODO: Obtain this clusterrole info from a Configmap or other source, rather than hard code
+	clusterRole := &rbacv1.ClusterRole{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "providerconfig-" + namespace.Name,
```
Comment (samidbb) [inline controllers/namespace_controller.go]: I think this should be my-namespace-aws to follow the convention we use in our compositons. 
The code should probably be like this Name: namespace.Name + "-aws",

File: controllers/namespace_controller.go [hunk 2]
```diff
@@ -92,77 +117,157 @@ func (r *NamespaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		},
 	}
 
-	clusterRoleBinding := &rbac.ClusterRoleBinding{
+	// TODO: Obtain this clusterrolebinding info from a Configmap or other source, rather than hard code
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{
 		ObjectMeta: metav1.ObjectMeta{
 			Name: "providerconfig-" + namespace.Name,
```
Comment (samidbb) [inline controllers/namespace_controller.go]: same as line 83

File: controllers/namespace_controller.go (hunk 1 above)
Comment (rifisdfds) [reply controllers/namespace_controller.go]: Makes sense, I have changed it

File: controllers/namespace_controller.go (hunk 2 above)
Comment (rifisdfds) [reply controllers/namespace_controller.go]: Makes sense, I have changed it

**Review by samidbb — APPROVED** [review]

**Pull request merged**